### PR TITLE
[JENKINS-33577] Demonstration test

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-33577](https://issues.jenkins-ci.org/browse/JENKINS-33577)

@reviewbybees esp. @apemberton

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     </licenses>
 
     <properties>
-        <workflow.version>1.1</workflow.version>
+        <workflow.version>1.4.2</workflow.version>
```
Comment (jglick) [inline pom.xml]: `build` did not return an appropriate value in the ancient 1.1 release. 1.4.2 is the newest compatible with 1.580.x.

File: pom.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <properties>
       <jenkins.version>1.580.1</jenkins.version>
       <java.level>6</java.level>
-      <workflow.version>1.1</workflow.version>
+      <workflow.version>1.4.2</workflow.version>
```
Comment (jglick) [inline pom.xml]: `build` did not return an appropriate value in the ancient 1.1 release. 1.4.2 is the newest compatible with 1.580.x.

**Comment by jglick** [comment]
@ikedam, @apemberton ping

**Comment by ikedam** [comment]
@jglick 
The test looks good.
It demonstrates that `SpecificBuildSelector` and `build(p)` provides a feature like `DownstreamBuildSelector`.

But I don't think this change needs to be merged as this doesn't provide any fixes or new features of copyartifact, but contains changes for dependencies that isn't important for copyartifact itself.

Do you mean upgrading copyartifact-plugin should force also to upgrade pipeline-plugin?

**Comment by jglick** [comment]
Well [1.4.2 is a rather old version](https://github.com/jenkinsci/pipeline-plugin/blob/master/CHANGES.md#142-jul-21-2015). Anyway this is only `test` scope so it has no effect on users.

**Comment by ikedam** [comment]
> Anyway this is only test scope so it has no effect on users.

You're absolutely right.
Sorry, I misunderstood that copyartifact has an optional dependency to pipeline.
I found core provides everything necessary to support pipeline.

I think this is ready to merge.
Though this looks a test for pipiline's `build` method rather than a test for copyartifact, I think it makes sense also as a test to use `selector` with pipilene.

**Pull request merged**